api/src/main/java/org/apache/iceberg/DeletedRowsScanTask.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+
+/**
+ * A scan task for deleted data records generated by adding delete files to the table.
+ */
+public interface DeletedRowsScanTask extends ChangelogScanTask, ContentScanTask<DataFile> {
```

Comment:
Let me make sure I understand correctly...

Say I have a table/partition with files `A`, `B`, and `C`. Then in `S1` I add a position delete file, `X`, that removes from `B`, and finally add an equality delete file, `Y` in `S2`. For changes from `S1` to `S2` (inclusive), this will produce:

* `DeletedRowsScanTask(file=B, added-deletes=[X], existing-deletes=[], snapshot=S1)`
* `DeletedRowsScanTask(file=A, added-deletes=[Y], existing-deletes=[], snapshot=S2)`
* `DeletedRowsScanTask(file=B, added-deletes=[Y], existing-deletes=[X], snapshot=S2)`
* `DeletedRowsScanTask(file=C, added-deletes=[Y], existing-deletes=[], snapshot=S2)`

Is that correct? An example in Javadoc would be helpful.

Reply:
I think a quick example in each of these java docs would be helpful.  Like suggested by @rdblue above, just what is the expected output of a given task type.

For example here
```
A scan task for producing records which were deleted because new delete files were added to the table.

Given File A with rows (1 - 10) IN S1 with row 5 marked as deleted by a delete file D1.
Given S2 with a new Delete File D2 deleting row 6

DRowScanTask (S2) returns {
files : (a)
addedDeletes: (D2)
existingDeletes: (D1)
}
Which should return row 6 because that row is newly marked as deleted by S2
```
Or something like that

Reply:
Added examples to all task types.